pkg/cli: Migrate to pgx v5

spilchen opened the pull request:
This updates the pkg/cli packages to use version 5 of the jackc/pgx library. While most changes were straightforward, transitioning from v4 to v5 required a few API updates:
- ConnInfo was renamed to TypeMap. Any references to connInfo were updated to typeMap for consistency.
- pgtype.Numeric can now be converted to []byte using MarshalJSON. Previously, this was done using EncodeText, which has been removed in v5.

Epic: none
Release note: none

Informs #137595

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/139667)
<!-- Reviewable:end -->

rafiss reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139667)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi, @arjunmahishi, @kyle-a-wong, and @spilchen)

___
*[`pkg/cli/BUILD.bazel` line 488 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/139667#-OHP3ee9AI4aXFFJDm4p:-OHP3eeA9nK-A9tvEbYc:bkgv94k) ([raw file](https://github.com/cockroachdb/cockroach/blob/7b8c507bd8f76c33356a7e6d1bbad5c7a12b7e37/pkg/cli/BUILD.bazel#L488)):*
> ```Starlark
>         # TODO(#81375)
>         #"//pkg/security/securitytest",
>         "//pkg/sql/randgen",
> ```

i wonder if we would be able to add pgx/v4 as a disallowed import here, so it doesn't get imported in the future


<!-- Sent from Reviewable.io -->

spilchen committed:
pkg/cli: Migrate to pgx v5

This updates the pkg/cli packages to use version 5 of the jackc/pgx
library. While most changes were straightforward, transitioning from v4
to v5 required a few API updates:
- ConnInfo was renamed to TypeMap. Any references to connInfo were
  updated to typeMap for consistency.
- pgtype.Numeric can now be converted to []byte using MarshalJSON.
  Previously, this was done using EncodeText, which has been removed in
  v5.

Epic: none
Release note: none

Informs #137595

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139667)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi, @arjunmahishi, @kyle-a-wong, and @rafiss)

___
*[`pkg/cli/BUILD.bazel` line 488 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/139667#-OHP3ee9AI4aXFFJDm4p:-OHPElEk0D6CEihLCtOL:bo6ca9i) ([raw file](https://github.com/cockroachdb/cockroach/blob/7b8c507bd8f76c33356a7e6d1bbad5c7a12b7e37/pkg/cli/BUILD.bazel#L488)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

i wonder if we would be able to add pgx/v4 as a disallowed import here, so it doesn't get imported in the future
</blockquote></details>

Cool. I didn't know that we could test for this. I couldn't add all of the pgx v4 imports. There were some imported through other packages that still need to be migrated. Here are two such cases:

❯ bazel query 'somepath(@//pkg/cli:cli, @com_github_jackc_pgtype//:pgtype)'
//pkg/cli:cli
//pkg/workload/tpcc:tpcc
@com_github_jackc_pgtype//:pgtype

❯ bazel query 'somepath(@//pkg/cli:cli, @com_github_jackc_pgconn//:pgconn)'
//pkg/cli:cli
//pkg/workload/tpcc:tpcc
@com_github_jackc_pgconn//:pgconn


<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR!

bors r+
